src/reformat_libyuv.c

```diff
@@ -975,85 +973,84 @@ avifResult avifImageYUVToRGBLibYUV(const avifImage * image, avifRGBImage * rgb,
-                                              image->yuvRowBytes[vPlaneIndex],
+            libyuvResult = lcf.yuvToRgbMatrix(image8->yuvPlanes[AVIF_CHAN_Y],
+                                              image8->yuvRowBytes[AVIF_CHAN_Y],
+                                              image8->yuvPlanes[uPlaneIndex],
+                                              image8->yuvRowBytes[uPlaneIndex],
+                                              image8->yuvPlanes[vPlaneIndex],
+                                              image8->yuvRowBytes[vPlaneIndex],
                                               rgb->pixels,
                                               rgb->rowBytes,
                                               matrix,
-                                              image->width,
-                                              image->height);
+                                              image8->width,
+                                              image8->height);
         } else if (lcf.yuvaToRgbMatrix != NULL) {
-            libyuvResult = lcf.yuvaToRgbMatrix(image->yuvPlanes[AVIF_CHAN_Y],
-                                               image->yuvRowBytes[AVIF_CHAN_Y],
-                                               image->yuvPlanes[uPlaneIndex],
-                                               image->yuvRowBytes[uPlaneIndex],
-                                               image->yuvPlanes[vPlaneIndex],
-                                               image->yuvRowBytes[vPlaneIndex],
-                                               image->alphaPlane,
-                                               image->alphaRowBytes,
+            libyuvResult = lcf.yuvaToRgbMatrix(image8->yuvPlanes[AVIF_CHAN_Y],
+                                               image8->yuvRowBytes[AVIF_CHAN_Y],
+                                               image8->yuvPlanes[uPlaneIndex],
+                                               image8->yuvRowBytes[uPlaneIndex],
+                                               image8->yuvPlanes[vPlaneIndex],
+                                               image8->yuvRowBytes[vPlaneIndex],
+                                               image8->alphaPlane,
+                                               image8->alphaRowBytes,
                                                rgb->pixels,
                                                rgb->rowBytes,
                                                matrix,
-                                               image->width,
-                                               image->height,
+                                               image8->width,
+                                               image8->height,
                                                /*attenuate=*/0);
             *alphaReformattedWithLibYUV = AVIF_TRUE;
         }
-        if (inputIsHighBitDepth) {
-            avifImageFreePlanes(&image8, AVIF_PLANES_YUV);
+        if (image8 != image) {
```

Comment:
Nit: If we test the equivalent condition `image8 == &image_downshifted_to_8bit`, it will further reinforce why `image_downshifted_to_8bit` needs to be freed.

Reply:
Done in https://github.com/AOMediaCodec/libavif/pull/1321.